### PR TITLE
[NFC] Move DxilValidation to a dedicated folder.

### PR DESCRIPTION
**Pull request by python3kgae**
Move DxilValidation out of HLSL.
Also move code to validate dxil container into DxilContainerValidation.cpp from DxilValidation.cpp.

This is a preparatory step for #6817.

### CODE REVIEW

File: lib/DxilValidation/DxilValidationUtils.h [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  const Type *HandleTy;
+  const DataLayout &DL;
+  DebugLoc LastDebugLocEmit;
+  ValidationRule LastRuleEmit;
+  std::unordered_set<Function *> entryFuncCallSet;
+  std::unordered_set<Function *> patchConstFuncCallSet;
+  std::unordered_map<unsigned, bool> UavCounterIncMap;
+  std::unordered_map<Value *, unsigned> HandleResIndexMap;
+  // TODO: save resource map for each createHandle/createHandleForLib.
+  std::unordered_map<Value *, DxilResourceProperties> ResPropMap;
```
Comment (bob80905) [inline lib/DxilValidation/DxilValidationUtils.h:80]: Looks like `DxilResourceProperties` is an incomplete type, never gets defined after its declaration on line 51.

Comment (python3kgae) [reply lib/DxilValidation/DxilValidationUtils.h:80]: Fixed.

File: lib/DxilValidation/DxilContainerValidation.cpp [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+  const char *pName = nullptr;
+  switch (SigKind) {
+  case hlsl::DXIL::SignatureKind::Input:
+    pName = "Program Input Signature";
+    break;
+  case hlsl::DXIL::SignatureKind::Output:
+    pName = "Program Output Signature";
+    break;
+  case hlsl::DXIL::SignatureKind::PatchConstOrPrim:
+    if (ValCtx.DxilMod.GetShaderModel()->GetKind() == DXIL::ShaderKind::Mesh)
```
Comment (bob80905) [inline lib/DxilValidation/DxilContainerValidation.cpp:131]: Should we check for node shaders here? I think mesh nodes can have patch constant functions right?

Comment (python3kgae) [reply lib/DxilValidation/DxilContainerValidation.cpp:131]: We don't have a Signature part for node shaders.

**Review by bob80905 — APPROVED** [review]

**Comment by damyanp** [comment]
Is this code really moved or is it copied from somewhere?  I see a lot of code being added, but not removed from anywhere.

**Comment by python3kgae** [comment]
> Is this code really moved or is it copied from somewhere? I see a lot of code being added, but not removed from anywhere.

The code is moved around and spited into different files.
Many removing happened in DxilValidation.cpp and placed into DxilContainerValidation.cpp.

File: utils/hct/hctdb_instrhelp.py [hunk 3]
```diff
@@ -1941,7 +1941,6 @@ def RunCodeTagUpdate(file_path):
             "lib/DXIL/DxilShaderModel.cpp",
             "include/dxc/DXIL/DxilConstants.h",
             "include/dxc/DXIL/DxilShaderModel.h",
-            "include/dxc/HLSL/DxilValidation.h",
```
Comment (damyanp) [inline utils/hct/hctdb_instrhelp.py:1944]: This doesn't need to be updated to look at the new location?

**Review by damyanp — APPROVED** [review]

File: utils/hct/hctdb_instrhelp.py (hunk 3 above)
Comment (python3kgae) [reply utils/hct/hctdb_instrhelp.py:1944]: There's no <py::lines left in DxilValidation.h after https://github.com/microsoft/DirectXShaderCompiler/commit/6cf0a6f7df67c72835d88659d410a43990307fc3

**Pull request merged**